test/specs/parsing/axes/AncestorAxis.tests.ts

```diff
@@ -14,50 +17,89 @@ beforeEach(() => {
 
 describe('ancestor', () => {
 	it('parses ancestor::', () => {
-		jsonMlMapper.parse([
-			'someParentElement',
-			['someElement', { someAttribute: 'someValue' }]
-		], documentNode);
-		chai.assert.deepEqual(evaluateXPathToNodes('ancestor::someParentElement', documentNode.documentElement.firstChild), [documentNode.documentElement]);
+		jsonMlMapper.parse(
+			['someParentElement', ['someElement', { someAttribute: 'someValue' }]],
+			documentNode
+		);
+		chai.assert.deepEqual(
+			evaluateXPathToNodes(
+				'ancestor::someParentElement',
+				documentNode.documentElement.firstChild
+			),
+			[documentNode.documentElement]
+		);
 	});
 
 	it('correctly sets contextSequence', () => {
-		jsonMlMapper.parse([
-			'someParentElement',
-			['someElement', { someAttribute: 'someValue' }]
-		], documentNode);
-		chai.assert.deepEqual(evaluateXPathToStrings('ancestor-or-self::*/position()', documentNode.documentElement.firstChild), ['1', '2']);
+		jsonMlMapper.parse(
+			['someParentElement', ['someElement', { someAttribute: 'someValue' }]],
+			documentNode
+		);
+		chai.assert.deepEqual(
+			evaluateXPathToStrings(
+				'ancestor-or-self::*/position()',
+				documentNode.documentElement.firstChild
+			),
+			['1', '2']
+		);
+	});
+
+	it('passes buckets for ancestor', () => {
+		jsonMlMapper.parse(['parentElement', ['childElement']], documentNode);
+
+		const childNode = documentNode.firstChild.firstChild;
+
+		const testDomFacade: IDomFacade = {
+			getParentNode: (node: slimdom.Node, bucket: string | null) => {
+				chai.assert.include(getBucketsForNode(node.parentNode), bucket);
```

Comment:
I agree with this test. I like how it does not explicitly depend on the format of a bucket, but just tests whether it matches.